src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -301,11 +301,7 @@ private function parseXmlAttributes(\DOMNode $node)
         $data = array();
 
         foreach ($node->attributes as $attr) {
-            if (ctype_digit($attr->nodeValue)) {
-                $data['@'.$attr->nodeName] = (int) $attr->nodeValue;
-            } else {
-                $data['@'.$attr->nodeName] = $attr->nodeValue;
-            }
+            $data['@'.$attr->nodeName] = $attr->nodeValue;
```

Comment:
@sstok this is a BC break, earlier would return as a string as `ctype_digit('-321312') === false`